sensor/common/networkflow/manager/manager_impl_test.go

```diff
@@ -179,7 +193,521 @@ func (suite *NetworkflowManagerTestSuite) TestAddNoOriginator() {
 	h.connectionsSequenceID = sequenceID
 
 	err := h.Process(networkInfoOpen, nowTimestamp, sequenceID)
-	suite.NoError(err)
+	s.NoError(err)
+
+	s.Len(h.endpoints, 1)
+}
+
+//endregion
+
+//region networkFlowManager tests
+
+func (s *NetworkFlowManagerTestSuite) TestEnrichConnection() {
+	mockCtrl := gomock.NewController(s.T())
+	m, mockEntityStore, mockExternalSrc, _ := createManager(mockCtrl)
+	containerID := "id"
+	cases := map[string]struct {
+		connPair                    *connectionPair
+		enrichConnections           map[networkConnIndicator]timestamp.MicroTS
+		expectEntityLookupContainer expectFn
+		expectEntityLookupEndpoint  expectFn
+		expectExternalLookup        expectFn
+		expectedIndicator           *networkConnIndicator
+		expectedConnection          *connection
+		expectedStatus              *connStatus
+	}{
+		"Rotten connection": {
+			connPair:                    createConnectionPair(true, true, false, timestamp.Now().Add(-maxContainerResolutionWaitPeriod*2)),
+			expectEntityLookupContainer: expectEntityLookupContainerHelper(mockEntityStore, 1, clusterentities.ContainerMetadata{}, false),
```

Comment:
I am wondering whether the code without helpers would look better?

```suggestion
			expectEntityLookupContainer: func() {
				mockEntityStore.EXPECT().LookupByContainerID(gomock.Any()).Times(1).DoAndReturn(func(_ any) (clusterentities.ContainerMetadata, bool) {
					return clusterentities.ContainerMetadata{}, false
				})
			}
```

Reply:
I did that at the beginning at it seemed a little bit difficult to read IMO. I can try to see if I like it now though :)

Reply:
I tried it in the `TestEnrichConenction` (see [ecccc60](https://github.com/stackrox/stackrox/pull/6959/commits/ecccc60f9d5c978930334fc29a8c20fcec00989a)). I don't have a strong opinion regarding this. On one hand I think it looks better because you don't have to check what the helper functions do, but on the other hand I think it is too verbose somehow. If you think it looks better I'll change it in the rest of the tests, otherwise I'll revert the last commit 🙂

Reply:
I don't have strong opinion, but I think it looks better without the helpers = less scrolling to understand what is going on. 
We can also ask other colleagues for opinion if you need stronger signals :)

Reply:
I prefer the code with the helpers. When looking at the test case, I feel like the code with the helpers shows just the important part, and it makes it easier to locate what changes from one test to the other. 

This whole thing may come down to personal preference, but since there are no strong signals, I'll leave mine here:

When it's a single test, I don't mind being more explicit, but if we have dozens of cases with the same boilerplate `.EXPECT()` parameters, then I'd prefer to move them to a helper function.

Take this one for example:

```go
"Incoming connection with successful lookup should not return a networkConnIndicator": {
	connPair:            createConnectionPair().incoming(),
	enrichedConnections: make(map[networkConnIndicator]timestamp.MicroTS),
	expectEntityLookupContainer: func() {
		mockEntityStore.EXPECT().LookupByContainerID(gomock.Any()).Times(1).DoAndReturn(func(_ any) (clusterentities.ContainerMetadata, bool) {
			return clusterentities.ContainerMetadata{
				DeploymentID: srcID,
			}, true
		})
	},
	expectEntityLookupEndpoint: func() {
		mockEntityStore.EXPECT().LookupByEndpoint(gomock.Any()).Times(1).DoAndReturn(func(_ any) []clusterentities.LookupResult {
			return []clusterentities.LookupResult{
				{
					Entity: networkgraph.Entity{
						ID: dstID,
					},
				},
			}
		})
	},
	expectedStatus: &connStatus{
		used: true,
	},
},
```


The things I'm interested in are:
- `createConnectionPair().incoming()`
- `clusterentities.ContainerMetadata` with `srcID`
- One `clusterentities.LookupResult` with `dstID` 
- And the assertion `connStatus.used == true`. 

But I have to squint my eyes and mentally remove all the boilerplate to see what's being changed from one test case to another. The alternative is much nicer, IMO:

```go
"Incoming connection with successful lookup should not return a networkConnIndicator": {
	connPair:            createConnectionPair().incoming(),
	enrichedConnections: make(map[networkConnIndicator]timestamp.MicroTS),
	expectEntityLookupContainer: expectEntityLookupContainerHelper(mockEntityStore, 1, clusterentities.ContainerMetadata{
		DeploymentID: srcID,
	}, true),
	expectEntityLookupEndpoint: expectEntityLookupEndpointHelper(mockEntityStore, 1, []clusterentities.LookupResult{
		{
			Entity: networkgraph.Entity{
				ID: dstID,
			},
		},
	}),
	expectedStatus: &connStatus{
		used: true,
	},
}
```